### PR TITLE
Implemented automatic release version and changelog generation

### PR DESCRIPTION
**Pull request by fragsalat**
Pull request for issue: #77 

This PR introduces a automatic tag and GitHub release version to be created when the master or release branch is build. So if you push to any other branch just the styleguide gets built but if a PR will be merged to the master the build also bumps the package version, generates a changelog, updates the dist folder, creates a new git tag and creates a new release on GitHub.

Circle ci deployment
circleci-release

Resulting version
github-release

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "babel-preset-react": "6.24.1",
     "babel-preset-stage-1": "6.24.1",
     "chai": "3.5.0",
-    "conventional-changelog": "1.1.3",
+    "conventional-github-releaser": "^1.1.9",
```
Comment (fokusferit) [inline package.json]: I think fix version is better so we are in full control :D

File: package.json [hunk 2]
```diff
@@ -70,6 +70,7 @@
     "gulp": "3.9.1",
     "gulp-babel": "6.1.2",
     "gulp-bump": "2.7.0",
+    "gulp-conventional-changelog": "^1.1.3",
```
Comment (fokusferit) [inline package.json]: same here

File: package.json [hunk 3]
```diff
@@ -56,7 +56,7 @@
     "babel-preset-react": "6.24.1",
     "babel-preset-stage-1": "6.24.1",
     "chai": "3.5.0",
-    "conventional-changelog": "1.1.3",
+    "conventional-github-releaser": "^1.1.9",
     "css-loader": "0.26.4",
     "del": "2.2.2",
     "documentation": "4.0.0-beta.19",
```
Comment (fokusferit) [inline package.json]: I would let this stay at beta.18 as long as you didn't push your documentationjs updates.

Comment (fragsalat) [inline package.json]: I created already the pr^^

Comment (fragsalat) [inline package.json]: Fail, saw right now that I didn't created the PR^^
So I will in 1 Minute

File: package.json (hunk 2 above)
Comment (fragsalat) [inline package.json]: Will be fixed with the documentation pr

File: package.json (hunk 1 above)
Comment (fragsalat) [inline package.json]: Will be fixed with the documentation pr

File: build/args.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
-var yargs = require('yargs');
+const yargs = require('yargs');
 
-var argv = yargs.argv;
-var validBumpTypes = 'major|minor|patch|prerelease'.split('|');
-var bump = (argv.bump || 'patch').toLowerCase();
+const argv = yargs.argv;
+const validBumpTypes = 'major|minor|patch|prerelease'.split('|');
+const bump = (argv.bump || 'patch').toLowerCase();
```
Comment (fokusferit) [inline build/args.js]: Does it mean 'patch' is default if not set different ? :D

Comment (fragsalat) [inline build/args.js]: Yes

**Pull request merged**